src/plugins/data_source_management/public/components/utils.ts

```diff
@@ -78,6 +79,41 @@ export async function setFirstDataSourceAsDefault(
   }
 }
 
+export function getFilterDataSources(
+  dataSources: Array<SavedObject<DataSourceAttributes>>,
+  filter?: (dataSource: SavedObject<DataSourceAttributes>) => boolean
+) {
+  return filter ? dataSources.filter((ds) => filter!(ds)) : dataSources;
+}
+
+export function getDefaultDataSource(
+  dataSources: Array<SavedObject<DataSourceAttributes>>,
+  LocalCluster: DataSourceOption,
+  uiSettings?: IUiSettingsClient,
+  hideLocalCluster?: boolean,
+  defaultOption?: DataSourceOption[]
+) {
+  let defaultShowingDataSource: DataSourceOption;
+
+  if (defaultOption && dataSources.find((dataSource) => dataSource.id === defaultOption?.[0].id)) {
```

Comment:
May be we can define the those value at the beginning, make the logic more clear
````
if (defaultDataSource) {
    ...
  } else if (uiSettingsDataSource) {
    ....
  } else if (!hideLocalCluster) {
    ....
  } else {
    ...
  }
````

Reply:
@zhyuanqi @Flyingliuhub 

think about adopting early return pattern 

https://gomakethings.com/the-early-return-pattern-in-javascript/